### PR TITLE
refactor: replace verbose help definitions with text-based format

### PR DESCRIPTION
**Pull request by gnodet**
## Summary
- Add `HelpOverlay.parse(String)` method that converts a simple markdown-like text format (`## Section` headers, 2+ space key/description delimiter) into `List<Section>`
- Add shared help text constants (`NAVIGATION_HELP`, `SEARCH_HELP`, `GENERAL_STANDALONE_HELP`, `GENERAL_STANDALONE_SAVE_HELP`) to `ToolPanel`
- Convert all 9 TUI files (`PomTui`, `TreeTui`, `ConflictsTui`, `DependenciesTui`, `UpdatesTui`, `SearchTui`, `AuditTui`, `AlignTui`, `ReactorUpdatesTui`) from verbose `new Section(..., List.of(new Entry(...)))` to `HelpOverlay.parse("""...""")` text blocks

## Test plan
- [x] `./mvnw compile -B` passes
- [x] `./mvnw test -B` passes
- [ ] Manual verification: press `h` in each tool to confirm help renders correctly
- [ ] CI green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Refactor**
  * Help overlays throughout the UI are now created by parsing multi-line help text instead of manual object assembly, preserving content, order, and behavior.
* **New Tests**
  * Added unit tests validating multi-section help parsing, key/description extraction, and edge-case handling.
* **Chore**
  * Introduced shared multi-line help text constants for navigation, search, and general/standalone help.

### CODE REVIEW

File: pilot-core/src/main/java/eu/maveniverse/maven/pilot/AlignTui.java [hunk 1]
```diff
@@ -566,54 +566,48 @@ void writeCrossPomChanges() {
...
-                                new HelpOverlay.Entry("q / Esc", "Quit"))));
+        sections.set(sections.size() - 1, HelpOverlay.parse("""
+                ## Keys
+                ↑ / ↓           Move between options
+                ← / → / Enter  Cycle through option values
+                p               Preview the POM changes as a diff
+                w               Apply alignment and write to POM
+                h               Toggle this help screen
+                q / Esc         Quit
+                """).get(0));
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/main/java/eu/maveniverse/maven/pilot/AlignTui.java]: _Claude Code on behalf of Guillaume Nodet_

Fixed — added defensive `isEmpty()` check before `get(0)` in both AlignTui and ConflictsTui.

File: pilot-core/src/main/java/eu/maveniverse/maven/pilot/DependenciesTui.java [hunk 2]
```diff
@@ -526,43 +526,37 @@ public List<Span> keyHints() {
...
+
+                ## Colors
+                yellow          Issue flag — unused or undeclared dependency
+                cyan            Header and view tab indicators
+                dim             Informational / secondary text
+
+                ## Dependencies Actions
+                ↑ / ↓           Move selection up / down
+                ← / →           Switch Declared / Transitive view
+                x / Enter       Remove selected (Declared view)
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/main/java/eu/maveniverse/maven/pilot/DependenciesTui.java:554]: _Claude Code on behalf of Guillaume Nodet_

This is correct — `DependenciesTui` overrides `subViewCount()` returning 2, so the shell handles `← / →` for sub-view switching in panel mode. The standalone `buildHelpStandalone()` documents `Tab` instead since standalone mode wires that key directly.

File: pilot-core/src/main/java/eu/maveniverse/maven/pilot/HelpOverlay.java [hunk 3]
```diff
@@ -48,6 +48,36 @@ record Entry(String key, String description) {}
...
+                } else {
+                    currentEntries.add(new Entry("", trimmed));
+                }
+            }
+        }
+        if (currentTitle != null) {
+            sections.add(new Section(currentTitle, currentEntries));
+        }
+        return sections;
+    }
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/main/java/eu/maveniverse/maven/pilot/HelpOverlay.java:79]: _Claude Code on behalf of Guillaume Nodet_

Done — 7 tests added in `HelpOverlayTest` covering multiple sections, description-only entries, blank lines, empty input, stray lines, concatenated text blocks, and split behavior.

File: pilot-core/src/main/java/eu/maveniverse/maven/pilot/ToolPanel.java [hunk 4]
```diff
@@ -56,6 +56,19 @@ public abstract class ToolPanel {
     /** Help overlay for standalone mode (shell has its own for panel mode). */
     protected final HelpOverlay helpOverlay = new HelpOverlay();
 
+    protected static final String SEARCH_HELP = """
+            ## Search
+            /               Enter search mode — type to search
+            n / N           Next / previous search match
+            Esc             Clear search or quit
+            """;
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/main/java/eu/maveniverse/maven/pilot/ToolPanel.java:71]: _Claude Code on behalf of Guillaume Nodet_

Fixed — changed to "Exit search mode".

File: pilot-core/src/main/java/eu/maveniverse/maven/pilot/ToolPanel.java [hunk 5]
```diff
@@ -56,6 +56,19 @@ public abstract class ToolPanel {
...
+
+    protected static final String GENERAL_STANDALONE_HELP = """
+            ## General
+            h               Toggle this help screen
+            q / Esc         Quit
+            """;
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/main/java/eu/maveniverse/maven/pilot/ToolPanel.java:77]: _Claude Code on behalf of Guillaume Nodet_

Spacing varies intentionally — the parser only requires 2+ spaces and the rendered output aligns keys uniformly via `HelpOverlay.open()`. Matching column widths in source would be brittle since each section has different key lengths.

File: pilot-core/src/test/java/eu/maveniverse/maven/pilot/HelpOverlayTest.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                This is a description line without a key.
+                Another description line.
+                """);
+
+        assertThat(sections).hasSize(1);
+        assertThat(sections.get(0).entries()).hasSize(2);
+        assertThat(sections.get(0).entries().get(0).key()).isEmpty();
+        assertThat(sections.get(0).entries().get(0).description())
+                .isEqualTo("This is a description line without a key.");
+    }
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/test/java/eu/maveniverse/maven/pilot/HelpOverlayTest.java:68]: _Claude Code on behalf of Guillaume Nodet_

Fixed — added explicit assertion for the second description-only entry.

File: pilot-core/src/main/java/eu/maveniverse/maven/pilot/AuditTui.java [hunk 7]
```diff
@@ -1373,61 +1373,52 @@ static String normalizeLicense(String name, String url) {
...
+        sections.addAll(HelpOverlay.parse("""
+                ## Keys
+                ↑ / ↓           Move selection up / down
+                ← / →           Collapse / expand (By License view)
+                Tab             Switch between Licenses / By License / Vulns
+                m               Add selected dep to dependencyManagement
+                d               Preview POM changes as a unified diff
+                h               Toggle this help screen
+                q / Esc         Quit (prompts to save if modified)
+                """));
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/main/java/eu/maveniverse/maven/pilot/AuditTui.java:1422]: _Claude Code on behalf of Guillaume Nodet_

Fixed — added `s  Cycle scope filter: all → compile → runtime → test → provided` to the standalone Keys section. This was a pre-existing omission in the original code.

File: pilot-core/src/main/java/eu/maveniverse/maven/pilot/UpdatesTui.java [hunk 8]
```diff
@@ -512,43 +512,38 @@ public List<Span> keyHints() {
...
+                ## Actions
+                ↑ / ↓           Move selection up / down
+                Space           Toggle selection of current dependency
+                a / n           Select all / deselect all
+                Enter           Apply selected updates to POM
+                d               Preview POM changes as unified diff
+                s / S           Sort by column / reverse sort
+                /               Search dependencies by name
+                n / N           Next / previous search match
+                """);
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/main/java/eu/maveniverse/maven/pilot/UpdatesTui.java]: _Claude Code on behalf of Guillaume Nodet_

Fixed — removed the `/` and `n / N` entries. Confirmed `UpdatesTui.handleKeyEvent()` does not call `handleSearchInput()`, so search is not supported. This was a pre-existing bug in the original code.

File: pilot-core/src/main/java/eu/maveniverse/maven/pilot/ConflictsTui.java [hunk 9]
```diff
@@ -523,20 +517,19 @@ private void toggleDiffView() {
...
+                Enter / Space   Toggle dependency path details
+                a               Toggle between conflicts only / all groups
+                p               Pin resolved version in dependencyManagement
+                d               Preview POM changes as a unified diff
+                h               Toggle this help screen
+                q / Esc         Quit (prompts to save if modified)
+                """);
+        if (!parsed.isEmpty()) {
+            sections.set(sections.size() - 1, parsed.get(0));
+        }
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/main/java/eu/maveniverse/maven/pilot/ConflictsTui.java:532]: _Claude Code on behalf of Guillaume Nodet_

The `sections` list from `helpSections()` always returns a fixed 4-section list, so it cannot be empty. We already guard the `parsed` list with an `isEmpty()` check, which is the part that could theoretically fail. Adding a guard on `sections` would be dead code.

File: pilot-core/src/test/java/eu/maveniverse/maven/pilot/HelpOverlayTest.java [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+                ## Keys
+                a / b           Description with  internal  spaces
+                single-word-key  value
+                """);
+
+        assertThat(sections).hasSize(1);
+        assertThat(sections.get(0).entries().get(0).key()).isEqualTo("a / b");
+        assertThat(sections.get(0).entries().get(0).description()).isEqualTo("Description with  internal  spaces");
+        assertThat(sections.get(0).entries().get(1).key()).isEqualTo("single-word-key");
+    }
```
[the remark this answers is not in the document]
Comment (gnodet) [reply pilot-core/src/test/java/eu/maveniverse/maven/pilot/HelpOverlayTest.java:134]: _Claude Code on behalf of Guillaume Nodet_

Fixed — added `assertThat(sections.get(0).entries().get(1).description()).isEqualTo("value")`.
